File: src/javax/microedition/io/HttpConnectionImpl.java
package javax.microedition.io;

import java.io.DataInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.URL;

final class HttpConnectionImpl implements HttpConnection {
  private final HttpURLConnection conn;
  private InputStream inputStream;

  HttpConnectionImpl(String url) throws IOException {
    URL u = new URL(url);
    conn = (HttpURLConnection) u.openConnection();
  }

  @Override
  public void setRequestMethod(String method) throws IOException {
    conn.setRequestMethod(method);
  }

  @Override
  public void setRequestProperty(String key, String value) {
    conn.setRequestProperty(key, value);
  }

  @Override
  public int getResponseCode() throws IOException {
    return conn.getResponseCode();
  }

  @Override
  public String getRequestMethod() {
    return conn.getRequestMethod();
  }

  @Override
  public String getHeaderField(String name) {
    return conn.getHeaderField(name);
  }

  @Override
  public long getLength() {
    return conn.getContentLengthLong();
  }

  @Override
  public String getEncoding() {
    return conn.getContentEncoding();
  }

  @Override
  public String getType() {
    return conn.getContentType();
  }

  @Override
  public InputStream openInputStream() throws IOException {
    if (inputStream == null) {
      inputStream = conn.getInputStream();
    }
    return inputStream;
  }

  @Override
  public DataInputStream openDataInputStream() throws IOException {
    return new DataInputStream(openInputStream());
  }

  @Override
  public void close() throws IOException {
    if (inputStream != null) {
      try {
        inputStream.close();
      } finally {
        inputStream = null;
      }
    }
    conn.disconnect();
  }
}

File: docs/swingjs/j2s/javax/microedition/io/HttpConnectionImpl.js
(function(){var P$=Clazz.newPackage("javax.microedition.io"),I$=[[0,'java.net.URL','java.io.DataInputStream']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "HttpConnectionImpl", null, null, 'javax.microedition.io.HttpConnection');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['O',['conn','java.net.HttpURLConnection','inputStream','java.io.InputStream']]]

Clazz.newMeth(C$, 'c$$S',  function (url) {
;C$.$init$.apply(this);
var u=Clazz.new_($I$(1,1).c$$S,[url]);
this.conn=u.openConnection$();
}, 1);

Clazz.newMeth(C$, 'setRequestMethod$S',  function (method) {
this.conn.setRequestMethod$S(method);
});

Clazz.newMeth(C$, 'setRequestProperty$S$S',  function (key, value) {
this.conn.setRequestProperty$S$S(key, value);
});

Clazz.newMeth(C$, 'getResponseCode$',  function () {
return this.conn.getResponseCode$();
});

Clazz.newMeth(C$, 'getRequestMethod$',  function () {
return this.conn.getRequestMethod$();
});

Clazz.newMeth(C$, 'getHeaderField$S',  function (name) {
return this.conn.getHeaderField$S(name);
});

Clazz.newMeth(C$, 'getLength$',  function () {
return this.conn.getContentLengthLong$();
});

Clazz.newMeth(C$, 'getEncoding$',  function () {
return this.conn.getContentEncoding$();
});

Clazz.newMeth(C$, 'getType$',  function () {
return this.conn.getContentType$();
});

Clazz.newMeth(C$, 'openInputStream$',  function () {
if (this.inputStream == null ) {
this.inputStream=this.conn.getInputStream$();
}return this.inputStream;
});

Clazz.newMeth(C$, 'openDataInputStream$',  function () {
return Clazz.new_([this.openInputStream$()],$I$(2,1).c$$java_io_InputStream);
});

Clazz.newMeth(C$, 'close$',  function () {
if (this.inputStream != null ) {
try {
this.inputStream.close$();
} finally {
this.inputStream=null;
}
}this.conn.disconnect$();
});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v5');//Created 2026-06-25 15:29:03 Java2ScriptVisitor version 3.3.1-v5 net.sf.j2s.core.jar version 3.3.1-v5
